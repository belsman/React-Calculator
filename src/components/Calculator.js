import React from 'react';
import Header from './Header';
import App from './App';

export default () => {
  <>
    <Header />
    <div className="content">
      <div>
        <p>Let&apos;s do some math!</p>
      </div>
      <App />
    </div>
  </>;
};
