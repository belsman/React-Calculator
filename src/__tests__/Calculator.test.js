import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Calculator from '../components/Calculator';

describe('Calculator component test', () => {
  test("should renders 'Let's do some math!' text", () => {
    render(<Calculator />);
    const text = screen.getByText('Let\'s do some math!');
    expect(text).toBeInTheDocument();
  });

  test("should renders 'Let's do some math!' text", () => {
    render(<Calculator />);
    const node = screen.getByRole('calculator');
    expect(node).toBeInTheDocument();
  });

  test('should render the expected output', () => {
    const component = renderer.create(
      <Calculator />,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
