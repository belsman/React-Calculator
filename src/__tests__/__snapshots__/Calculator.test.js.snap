// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator component test should render the expected output 1`] = `
<div
  className="content"
  style={
    Object {
      "display": "flex",
      "justifyContent": "space-between",
    }
  }
>
  <div>
    <p
      style={
        Object {
          "fontSize": "1.2em",
        }
      }
    >
      Let's do some math!
    </p>
  </div>
  <div
    className="calculator"
  >
    <div
      id="display"
    >
      <div
        className="user-input"
      >
        0
      </div>
    </div>
    <div
      id="btn-panel"
    >
      <div
        className="btn-panel-row"
      >
        <button
          className="basic-operator"
          onClick={[Function]}
          type="button"
        >
          AC
        </button>
        <button
          className="basic-operator"
          onClick={[Function]}
          type="button"
        >
          +/-
        </button>
        <button
          className="basic-operator"
          onClick={[Function]}
          type="button"
        >
          %
        </button>
        <button
          className="basic-operator"
          onClick={[Function]}
          type="button"
        >
          รท
        </button>
      </div>
      <div
        className="btn-panel-row"
      >
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          7
        </button>
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          8
        </button>
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          9
        </button>
        <button
          className="basic-operator"
          onClick={[Function]}
          type="button"
        >
          x
        </button>
      </div>
      <div
        className="btn-panel-row"
      >
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          4
        </button>
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          5
        </button>
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          6
        </button>
        <button
          className="basic-operator"
          onClick={[Function]}
          type="button"
        >
          -
        </button>
      </div>
      <div
        className="btn-panel-row"
      >
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          1
        </button>
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          2
        </button>
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          3
        </button>
        <button
          className="basic-operator"
          onClick={[Function]}
          type="button"
        >
          +
        </button>
      </div>
      <div
        className="btn-panel-row"
      >
        <button
          className="flex-basis-50"
          onClick={[Function]}
          type="button"
        >
          0
        </button>
        <button
          className="flex-basis-25"
          onClick={[Function]}
          type="button"
        >
          .
        </button>
        <button
          className="basic-operator flex-basis-25"
          onClick={[Function]}
          type="button"
        >
          =
        </button>
      </div>
    </div>
  </div>
</div>
`;
